_posts/2024-08-09-contributing-to-eclipse-che-docs.adoc

```diff
@@ -0,0 +1,104 @@
+
+
+== Getting started
+
+Click the Developer Workspace button in the che-docs repository README file.
+This opens a clone of the repository in an Eclipse Che workspace hosted
+by Red Hat in the link:https://developers.redhat.com/developer-sandbox[Developer Sandbox].
+Alternatively, you can click link:https://workspaces.openshift.com#https://github.com/eclipse/che-docs[*right here*].
+
+== How does it work?
+
+What about the che-docs repository makes using an Eclipse Che CDE to contribute content so convenient? The following two elements:
+
+* link:https://eclipse.dev/che/docs/stable/end-user-guide/devfile-introduction/[devfile]
+* list of link:https://eclipse.dev/che/docs/stable/end-user-guide/microsoft-visual-studio-code-open-source-ide/#automating-installation-of-microsoft-visual-studio-code-extensions-at-workspace-startup[automatically installed Microsoft Visual Studio Code extensions]
+
+Let's look at the devfile first.
+The che-docs repository contains a simple link:https://github.com/eclipse-che/che-docs/blob/main/devfile.yaml[devfile]
+that defines the basic features of the che-docs workspace.
+It also includes a command for building a preview of your newly created content:
+
+[source, code]
+----
+# sh tools/preview.sh
+----
+
+This is helpful not only for checking the layout of the document but also for making sure that embedded images,
+links and code blocks are displayed correctly.
+
+By default, the che-docs workspace launches with Visual Studio Code - Open Source ("Code - OSS") as the IDE.
+This is relevant for the second file mentioned above.
+The che-docs repository features an link:https://github.com/eclipse-che/che-docs/blob/main/.vscode/extensions.json[extensions.json] file listing Visual Studio Code extensions
+you need for writing Eclipse Che content.
+When you launch the che-docs workspace for the first time,
+these extensions are installed automatically.
+You can start working immediately,
+with minimal additional setup required.
+
+image::/assets/img/contributing-to-eclipse-che-docs/extensions.png[Overview of automatically installed extensions.]
+
+== Getting started
+
+By launching the che-docs workspace, you're almost ready to go:
+
+* You have a clone of the che-docs repository
+* You have an IDE to write in.
+* You have required extensions installed.
+* You have a preview build script.
+
+There are only three things left to do.
+
+. Read the link:https://github.com/eclipse-che/che-docs/blob/main/CONTRIBUTING.adoc[contribution guidelines].
+. Read how to link:https://redhat-documentation.github.io/supplementary-style-guide/#technical-examples[document commands] for Eclipse Che documentation.
+. Update the grammar- and spelling-checking link:https://marketplace.visualstudio.com/items?itemName=ChrisChinchilla.vale-vscode[Vale extension] by running the following command in the terminal:
++
+[source, code]
+----
+# vale sync
```

Comment:
it looks like it is part of the https://github.com/eclipse-che/che-docs/blob/main/tools/preview.sh#L14
so this section with vale should probably be described with preview.sh

Reply:
it's still better to run it separately, because this way, errors get flagged immediately, before you run the preview script

Reply:
ok, but it should be explained in this case

Reply:
also, if running this command separately makes sense please contribute it to the devfile directly

Reply:
that's why I specifically mentioned it as a prerequisite :D should I add an explanation that the preview script syncs vale as well?

Reply:
if you want to leave it in this place let's add TIP / NOTE below:
`TIP: vale sync is part of devfile preview command, but it is recommended to run it that way because this and that`

Reply:
PR for adding `vale-sync` command to devfile - https://github.com/eclipse-che/che-docs/pull/2768